test/end-to-end/company.js

```diff
@@ -227,6 +228,290 @@ describe("Company endpoint", () => {
+                            "publishEndDate": (new Date(Date.now() + (DAY_TO_MS))).toISOString(),
+                            owner: test_company._id,
+                            ownerName: test_company.name,
+                            ownerLogo: test_company.logo
+                        })
+                    )).sort((a1, a2) => a1.publishDate > a2.publishDate));
+
+                const res = await test_agent
+                    .get(`/company/${test_company._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers.length).toEqual(CompanyConstants.offers.max_profile_visible);
+                offers.forEach((val, idx) => {
+                    expect(res.body.offers[idx].owner.toString()).toEqual(val.owner.toString());
+                    expect(res.body.offers[idx].ownerName.toString()).toEqual(val.ownerName.toString());
+                    expect(res.body.offers[idx].ownerLogo.toString()).toEqual(val.ownerLogo.toString());
+                });
+            });
+
+            test("should return the maximum number of offers to be displayed if there are more offers than it", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const offers = await Promise.all(Array(CompanyConstants.offers.max_profile_visible + 1).fill(null).map((_) =>
+                    Offer.create(
+                        generateTestOffer({
+                            "publishDate": (new Date(Date.now() - (DAY_TO_MS))).toISOString(),
+                            "publishEndDate": (new Date(Date.now() + (DAY_TO_MS))).toISOString(),
+                            owner: test_company_3._id,
+                            ownerName: test_company_3.name,
+                            ownerLogo: test_company_3.logo
+                        })
+                    )).sort((a1, a2) => a1.publishDate > a2.publishDate));
+
+                const res = await test_agent
+                    .get(`/company/${test_company_3._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers.length).toEqual(CompanyConstants.offers.max_profile_visible);
+                for (let idx = 0; idx < CompanyConstants.offers.max_concurrent; idx++) {
+                    const val = offers[idx];
+
+                    expect(res.body.offers[idx].owner.toString()).toEqual(val.owner.toString());
+                    expect(res.body.offers[idx].ownerName.toString()).toEqual(val.ownerName.toString());
+                    expect(res.body.offers[idx].ownerLogo.toString()).toEqual(val.ownerLogo.toString());
+                }
+            });
+
+            test("should return if a random user is unable to see hidden offers", async () => {
```

Comment:
Change this test description to something like "Should return no hidden offers when viewing company profile as unprivileged user"